SEQLinkage/Core.py

```diff
@@ -351,32 +388,12 @@ def __Haplotype(self, data):
         #
         # Per family haplotyping
         #
+        items = get_family_with_var(data)
+        with ProcessPoolExecutor(max_workers = 8) as executor:
+            inputs = executor.map(phasing_haps,repeat(data.chrom),items,[data.getFamVariants(item, style = "map") for item in items],[data.getFamSamples(item) for item in items])
+
         varnames,mafs,haplotypes = OrderedDict(),OrderedDict(),OrderedDict()
-        for item in data.families:
-            item_varnames, positions, item_mafs = data.getFamVariants(item, style = "map")
-            #env.dtest[self.name]['hapimp'][item] = [item,item_varnames, positions, item_mafs] #test line
-            if len(item_varnames) == 0:  #no variants in the family
-                for person in data.families[item]:
-                    data[person] = self.missings
-                continue
-            if self.maf_cutoff is not None:
-                keep_idx = item_mafs<self.maf_cutoff
-                if not keep_idx.any():
-                    for person in data.families[item]:
-                        data[person] = self.missings
-                    continue
-            if env.debug:
-                with env.lock:
-                    sys.stderr.write('\n'.join(['\t'.join(x) for x in data.getFamSamples(item)]) + '\n\n')
-            # haplotyping
-            with env.lock:
-                if not env.prephased:
-                    tmp_log_output=env.tmp_log + str(os.getpid()) + '.log'
-                    item_haplotypes = self.haplotyper.Execute(data.chrom, item_varnames, positions, data.getFamSamples(item), self.rsq, tmp_log_output)[0]
-                else:
-                    item_haplotypes = self.__PedToHaplotype(data.getFamSamples(item))
-                item_haplotypes = np.array(item_haplotypes)
-            #env.dtest[self.name]['hapimp'][item].append(item_haplotypes)#test line
+        for item,item_varnames,item_mafs,item_haplotypes in inputs:
```

Comment:
lots of code are deleted here. Can you tell me why they are removed (and make sure it's fine to do so?)

Reply:
I added parallelization to phasing haplotypes and reorganized the code.